src/main/scala/com/fulcrumgenomics/commons/io/CommandLineTool.scala

```diff
@@ -0,0 +1,205 @@
+import scala.util.Try
+
+trait CommandLineTool extends LazyLogging {
+
+  /** The name of the executable such as Rscript or gs. */
+  val Executable: String
+
+  /** Command used for the above executable.
+    *
+    * For example, to run `Rscript --version`:
+    * {{{
+    * scala> import com.fulcrumgenomics.commons.io._
+    * scala> val TestArgs = Seq("--version")
+    * scala> val Executable = "Rscript"
+    * scala> Rscript.execArgs(Executable +: TestArgs:_*)
+    * }}}
+    * */
+  val TestArgs: Seq[String]
+
+  /** Exception class that holds onto an executable's exit/status code. */
+  case class ToolException(status: Int) extends RuntimeException {
+    override def getMessage: String = s"$Executable failed with exit code $status."
+  }
+
+  /** Executes a command and returns the stdout.
+    *
+    * @param command the command to be executed
+    */
+  def execArgs(command: String*): Try[ListBuffer[String]] = {
+    Try {
+      // Read redirected stdout to a string
+      val process  = new ProcessBuilder(command: _*).redirectErrorStream(true).start()
+      val stdout   = ListBuffer[String]()
+      val sink     = new AsyncStreamSink(process.getInputStream, s => stdout.append(s))
+      val exitCode = process.waitFor()
+
+      sink.close()
+
+      require(exitCode == 0, s"Command did not execute successfully. Exit code: $exitCode")
+
+      stdout
+    }
+  }
+
+  /** Returns true if the tool is available and false otherwise. */
+  lazy val Available: Boolean = execArgs(Executable +: TestArgs: _*).isSuccess
+}
+
+/** Trait that indicates the executable can run scripts.*/
+trait ScriptRunner {
+  self: CommandLineTool =>
+
+  /** Suffix for scripts that can be run by this tool. */
+  val Suffix: String
+
+  /** Executes a script from the classpath if this executable is available.
+    *
+    * @param scriptResource the name of the script resource on the classpath
+    * @param args a variable list of arguments to pass to the script
+    * @throws Exception when we are unable to execute to this script on the classpath with the given arguments.
+    */
+  def execIfAvailable(scriptResource: String, args: String*): Unit =
+    try { if (Available) exec(scriptResource, args:_*) }
+    catch { case e: Exception => throw new Exception(s"Cannot execute script from $scriptResource with args:$args") }
+
+  /** Executes a script stored at a Path if the this executable is available.
+    *
+    * @param script Path of the script to be run
+    * @param args a variable list of arguments to pass to the script
+    * @throws Exception when we are unable to execute to this script on the classpath with the given arguments.
+    */
+  def execIfAvailable(script: Path, args: String*): Unit = execIfAvailable(script.toString, args: _*)
+
+  /** Executes a script from the classpath if this executable is available, raise an exception otherwise.
+    *
+    * @throws ToolException when the exit code from the called process is not zero.
```

Comment:
Whats the order of the scaladoc? `@param`, `@throws` and then `@return`? Make sure its consistant though this entire block.